src/Request/Request.php

```diff
@@ -0,0 +1,258 @@
+                compact('target')
+            );
+        }
+
+        $client = new Client();
+        $response = $client->request('GET', $url, ['sink' => $target,]);
+        return true;
+    }
+
+    /**
+     * Make a request to the Dashbord's internal API
+     *
+     * @param string $path API path (URL)
+     * @param array $options Options for the request
+     *   [string] method GET is default
+     *   [mixed]  data   Native PHP data structure (e.g. int, string array, or
+     *     simple object) to be sent along with the request. Will be encoded as
+     *     JSON for you.
+     * @return array
+     */
+    public function pagedRequest($path, array $options = [])
+    {
+        $limit = 100;
+        if (isset($options['limit'])) {
+            $limit = $options['limit'];
+        }
+
+        //$results is an associative array so we don't refetch
+        $results = [];
+        $finished = false;
+        $start = null;
+
+        while (!$finished) {
+            $paged_path = $path . '?limit=' . $limit;
+            if ($start) {
+                $paged_path .= '&start=' . $start;
+            }
+
+            $resp = $this->request($paged_path);
+
+            $data = $resp['data'];
+            if (count($data) > 0) {
+                if (count($data) < $limit) {
+                    $finished = true;
+                }
+                $start = end($data)->id;
+
+                //If the last item of the results has previously been received,
+                //that means there are no more pages to fetch
+                if (isset($results[$start])) {
+                    $finished = true;
+                    continue;
+                }
+
+                foreach ($data as $item) {
+                    $results[$item->id] = $item;
+                }
+            } else {
+                $finished = true;
+            }
+        }
+
+        $return = ['data' => array_values($results),];
+        return $return;
+    }
+
+    /**
+     * Simplified request method for Pantheon API
+     *
+     * @param string $path API path (URL)
+     * @param array $arg_options Options for the request
+     *   [string] method        GET is default
+     *   [mixed]  data          Native PHP data structure (e.g. int, string
+     *     array, or simple object) to be sent along with the request. Will
+     *     be encoded as JSON for you.
+     *   [boolean] absolute_url True if URL passed is to be treated as absolute
+     * @return array
+     * @throws TerminusException
+     */
+    public function request($path, $arg_options = [])
+    {
+        $default_options = [
+            'method' => 'get',
+            'absolute_url' => false,
+        ];
+        $options = array_merge($default_options, $arg_options);
+
+        $url = $path;
+        if ((strpos($path, 'http') !== 0) && !$options['absolute_url']) {
+            $url = sprintf(
+                '%s://%s:%s/api/%s',
+                $this->getConfig()->get('protocol'),
```

Comment:
Direct access to the `config` object via ConfigAwareInterface is fine where needed.  However, we should favor alternatives where possible:

- If applicable, getter/setter methods could be used, and the container could be set up to do a `withMethodCall` using config::get() to seed the initial value.  This would make the tests nicer, as they wouldn't need to construct or mock a config object.
- For Drush, I am planning on making a config system that uses the HookManager to look up command arguments and options from the config option in @hook init.  This will allow Drush users to make command-specific sections in their Drush config files that supplies default values to any parameter that is settable via a command argument or option.  This will take care of just about all of the config needs of commands.  Terminus could use this sort of facility as well.